packages/contracts/test/contracts/ovm/PartialStateManager.gas-metering-script.spec.ts

```diff
@@ -0,0 +1,250 @@
+        to: executionManager.address,
+        data: EMCallBytes,
+        gasLimit: GAS_LIMIT,
+      })
+    }
+  }
+
+  const getCumulativeSequencedGas = async (): Promise<number> => {
+    return hexStrToNumber(
+      (await executionManager.getCumulativeSequencedGas())._hex
+    )
+  }
+
+  const getChangeInCumulativeGas = async (
+    callbackConsumingGas: () => Promise<any>
+  ): Promise<{
+    internalToOVM: number
+    additionalExecuteTransactionOverhead: number
+  }> => {
+    const before: number = await getCumulativeSequencedGas()
+    const tx = await callbackConsumingGas()
+    const after: number = await getCumulativeSequencedGas()
+
+    const receipt = await executionManager.provider.getTransactionReceipt(
+      tx.hash
+    )
+
+    const change = after - before
+
+    return {
+      internalToOVM: change,
+      additionalExecuteTransactionOverhead:
+        hexStrToNumber(receipt.gasUsed._hex) - change,
+    }
+  }
+
+  before(async () => {
+    ;[wallet] = await ethers.getSigners()
+    walletAddress = await wallet.getAddress()
+    resolver = await makeAddressResolver(wallet)
+    ExecutionManager = await ethers.getContractFactory('ExecutionManager')
+    PartialStateManager = await ethers.getContractFactory('PartialStateManager')
+    SimpleStorage = await ethers.getContractFactory('SimpleStorage')
+
+    executionManager = await deployAndRegister(
+      resolver.addressResolver,
+      wallet,
+      'ExecutionManager',
+      {
+        factory: ExecutionManager,
+        params: [
+          resolver.addressResolver.address,
+          NULL_ADDRESS,
+          [
+            OVM_TX_BASE_GAS_FEE,
+            OVM_TX_MAX_GAS,
+            GAS_RATE_LIMIT_EPOCH_IN_SECONDS,
+            MAX_GAS_PER_EPOCH,
+            MAX_GAS_PER_EPOCH,
+          ],
+        ],
+      }
+    )
+
+    stateManager = (
+      await deployAndRegister(
+        resolver.addressResolver,
+        wallet,
+        'StateManager',
+        {
+          factory: PartialStateManager,
+          params: [resolver.addressResolver.address, walletAddress],
+        }
+      )
+    ).connect(wallet)
+
+    simpleStorageNative = await SimpleStorage.deploy()
+
+    await stateManager.insertVerifiedContract(walletAddress, walletAddress, 0)
+
+    await stateManager.insertVerifiedContract(
+      simpleStorageOVMAddress,
+      simpleStorageNative.address,
+      0
+    )
+  })
+
+  beforeEach(async () => {
+    // reset the single key value so that sstore costs are the same between tests
+    await stateManager.insertVerifiedStorage(
+      simpleStorageOVMAddress,
+      key,
+      '0x' + '00'.repeat(32)
+    )
+    // reset the sequential keys we set so sstore costs are the same between tests (new set vs update)
+    for (const aKey of multipleSequentialKeys) {
+      await stateManager.insertVerifiedStorage(
+        simpleStorageOVMAddress,
+        aKey,
+        '0x' + '00'.repeat(32)
+      )
+    }
+  })
+
+  it('setStorage', async () => {
```

Comment:
I think generally better to put each function into a separate `describe()` and then list out the cases within an `it()`.